src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -50,11 +50,20 @@ public function process(ContainerBuilder $container)
 
         $serializerDefinition = $container->getDefinition($this->serializerService);
         $serializerDefinition->replaceArgument(0, $normalizers);
+        $this->addDefaultContextParameterBinding($normalizers, $container);
 
         if (!$encoders = $this->findAndSortTaggedServices($this->encoderTag, $container)) {
             throw new RuntimeException(sprintf('You must tag at least one service as "%s" to use the "%s" service.', $this->encoderTag, $this->serializerService));
         }
 
         $serializerDefinition->replaceArgument(1, $encoders);
+        $this->addDefaultContextParameterBinding($encoders, $container);
+    }
+
+    private function addDefaultContextParameterBinding($services, $container)
+    {
+        foreach ($services as $service) {
+            $container->getDefinition($service)->setBindings(array('$defaultContext' => '%serializer.default_context%'));
```

Comment:
What should I add to check that the service has this binding? I get `Unused binding "$defaultContext" in service "serializer.normalizer.data_uri".` errors without checks but can't find out what would be best. Should I check the definition arguments?

Reply:
Yes you'll have to use reflection. Or maybe can you you call `$container->getDefinition($service)->setArgument('$defaultContext', '%serializer.default_context%'`);`. Maybe this one will not throw. Not sure what's the best practice (ping @nicolas-grekas).

Reply:
I think the best practice would be to:

A) For core services, wire it up manually in the services XML file - don't use binding.
B) For user-implemented normalizers, I can see 2 options:
1) Adding some sort of `DefaultContextAwareNormalizerInterface` with a `setDefaultContext()` method (we check for the interface, then add this call)
or
2) Wrap the normalizer context into a service (e.g. `serializer.default_context`) and document that people can DI this into their custom normalizers. We could even add an autowireable alias for this service so that simply type-hinting the service class will pass it to them.